### PR TITLE
Created a very generic read block that pulls all data from firebase

### PR DESCRIPTION
**Pull request by tlugger**
To create this I refactored the insert block to do a get request rather than put. It will return everything in the specified firebase collection as one signal. This does not yet have functionality to notify on database changes.

### CODE REVIEW

File: firebase_read_block.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    def process_signals(self, signals):
+        out_sigs = []
+        for sig in signals:
+            # The collection may be different for different signals,
+            # so we need to compute it inside the loop
+            collection = self._get_collection(sig)
+            try:
+                res = self._firebase.get(collection, None)
+            except:
+                self.logger.exception("Couldn't save signal")
```
Comment (hansmosh) [inline firebase_read_block.py]: Copy pasta

File: firebase_read_block.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from .firebase_base import FirebaseBase
+
+
+class FirebaseRead(FirebaseBase):
+
+    def process_signals(self, signals):
+        out_sigs = []
+        for sig in signals:
+            # The collection may be different for different signals,
+            # so we need to compute it inside the loop
```
Comment (hansmosh) [inline firebase_read_block.py]: This is definitely just my opinion, but this comment (and the one about enriching this signal) seem unnecessary. I don't like comments that just explain in english what the next line of code is doing, especially if that line of code isn't particularly confusing.

File: firebase_read_block.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from .firebase_base import FirebaseBase
+
+
+class FirebaseRead(FirebaseBase):
```
Comment (hansmosh) [inline firebase_read_block.py:4]: Lets get a unit test for this block.

**Review by hansmosh — CHANGES_REQUESTED** [review]

File: firebase_read_block.py (hunk 1 above)
Comment (tlugger) [reply firebase_read_block.py]: I'll fix the exception message to be more specific to the get request

File: firebase_read_block.py (hunk 2 above)
Comment (tlugger) [reply firebase_read_block.py]: I agree that the comment blocks aren't necessary. These were copied over from the insert block as it had very similar functionality. I'll remove them from this block but would you recommend that I take them out of the insert block as well?

**Review by hansmosh — APPROVED** [review]

**Pull request merged**